components/Login/LoginModal.js

```diff
@@ -31,11 +34,16 @@ class LoginModal extends React.Component {
         this.handleLoginButton = this.handleLoginButton.bind(this);
         this.handleSignupClick = this.handleSignupClick.bind(this);
         this.handleNoAccessClick = this.handleNoAccessClick.bind(this);
+        this.unmountTrap = this.unmountTrap.bind(this);
         this.signin = this.signin.bind(this);
         this.provider = '';
         this.isLoading = false;
+        this.state = {
+            activeTrap: this.props.LoginModalStore.activeTrap?this.props.LoginModalStore.activeTrap:false,
```

Comment:
Is there a reason to use the state? Instead just use this.props.LoginModalStore.activeTrap wherever this.state.activeTrap is used?

Reply:
It is needed to use state if you want mount/umount the trap. Changes in the state provoque it re-renders.

Reply:
Re AddDeck: 
The bug is the following:
When I am not logged in and click on "Add Deck", the LoginModal opens. When I then close it, the modal closes but the underlaying view keeps black.
Suggestion:
Please update the action loadAddDeck in order to have there the same mechanism as in you already changed code.
In general: Please make sure the usage of the LoginModal is everywhere the same.

Reply:
Now I undertand excactly the problem!! I only modified the code to open the modal from the sigin button, because I didn't know it was also used from another point!!. Thanks a lot for your review. I need toupdate also the call from the adddeck

Reply:
and from deckEditPanel, ResetPassword, UserProfileReview, UserProfileReviewUser, UserRegistration